File: src/main/java/leetcode/lists/topinterview/LeetCode171.java
package leetcode.lists.topinterview;

/**
 * @author wujingxinit@outlook.com
 * @date 2022/9/27 13:12
 */
public class LeetCode171 {

    //study
    public int titleToNumber(String columnTitle) {
        int ans = 0;
        for (int i = 0; i < columnTitle.length(); i++) {
            ans = ans * 26 + (columnTitle.charAt(i) - 'A' + 1);
        }
        return ans;
    }
}

File: go/src/leetcode/topinterview/LeetCode171.go
package topinterview

func titleToNumber(columnTitle string) int {
	if len(columnTitle) == 0 {
		return 0
	}
	number := 0
	for i, multiple := len(columnTitle)-1, 1; i >= 0; i-- {
		k := columnTitle[i] - 'A' + 1
		number += int(k) * multiple
		multiple *= 26
	}
	return number
}
